pkg/apiserver/handlers/webhook/validation_crd_test.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	admv1 "k8s.io/api/admission/v1"
+)
+
+const (
+	contentTypeJson = "application/json"
+	contentTypeHtml = "text/html"
+)
+
+type responseWriter struct {
```

Comment:
why are we defining our own object instead of using https://pkg.go.dev/net/http/httptest#ResponseRecorder which is typically used to test HTTP handlers?

Reply:
The reason is that we need to implement the failed cases, which are determined by `uid`, so we defined our own object.

Reply:
I don't think this is needed.
The only purpose it serves it to test the error case for https://github.com/antrea-io/antrea/blob/7866681e691294fb0a9aaeac564fdd43ca7e6ca6/pkg/apiserver/handlers/webhook/validation_crd.go#L78, which is not a really interesting error case to test for.

But if you insist on testing this case, I think you should just wrap the standard `httptest.ResponseRecorder`:
```go
type ResponseRecorder struct {
    *httptest.ResponseRecorder
    writeErr error
}

func NewRecorder(writeErr error) *ResponseRecorder {
    return &ResponseRecorder{ResponseRecorder: httptest.NewRecorder(), writeErr: error}
}

func (rr *ResponseRecorder) Write(b []byte) (int, error) {
    if (rr.writeErr != nil) return 0, rr.WriteErr
    return rr.ResponseRecorder.Write(b)
}
```
And only use this object for tests that actually need it.

Relying on the UID of the object to trigger an error in the Write function for your tests is confusing.

Reply:
Removed the "internal error" test case, as well as the `ResponseWriter`, using `httptest.Recorder` now.